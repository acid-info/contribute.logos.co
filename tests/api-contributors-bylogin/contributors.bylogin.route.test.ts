import { describe, it, expect, vi } from 'vitest'
import mocks, { MOCK_LOGIN, MOCK_ORGS } from './mocks'

function windowIso(days: number) {
  const until = new Date('2025-08-12T00:00:00Z')
  const since = new Date(until.getTime() - days * 24 * 60 * 60 * 1000)
  return { since: since.toISOString(), until: until.toISOString() }
}

vi.mock('@/lib/github', () => mocks)

describe('GET /api/contributors/[login]', () => {
  it('returns a contribution ledger for a user (mocked)', async () => {
    const login = MOCK_LOGIN
    const orgs = MOCK_ORGS.join(',')
    const { since, until } = windowIso(365)
    const url = `http://localhost/api/contributors/${login}?orgs=${encodeURIComponent(
      orgs
    )}&since=${encodeURIComponent(since)}&until=${encodeURIComponent(until)}&debug=true`
    const req = new Request(url)
    const { GET: GETByLogin } = await import('../../app/api/contributors/[login]/route')
    const res = await GETByLogin(req)
    expect(res.status).toBe(200)
    const json = (await res.json()) as any
    expect(json).toMatchSnapshot('contributors-bylogin-response')
  })
})
