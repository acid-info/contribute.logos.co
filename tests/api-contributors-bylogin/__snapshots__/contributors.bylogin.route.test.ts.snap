// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GET /api/contributors/[login] > returns a contribution ledger for a user (mocked) > contributors-bylogin-response 1`] = `
{
  "items": [
    {
      "date": "2025-07-03T00:00:00Z",
      "link": "https://github.com/acme-co/widget/pull/123#review-1",
      "repo": "acme-co/widget",
      "repoUrl": "https://github.com/acme-co/widget",
      "type": "REVIEW",
    },
    {
      "date": "2025-07-02T00:00:00Z",
      "link": "https://github.com/acme-co/widget/commit/abc",
      "repo": "acme-co/widget",
      "repoUrl": "https://github.com/acme-co/widget",
      "type": "COMMIT",
    },
    {
      "date": "2025-07-01T00:00:00Z",
      "link": "https://github.com/acme-co/widget/pull/123",
      "repo": "acme-co/widget",
      "repoUrl": "https://github.com/acme-co/widget",
      "type": "PR",
    },
  ],
  "login": "user-mock-123",
  "total": 3,
}
`;
